python/examples/property-selection.py

```diff
@@ -0,0 +1,121 @@
+
+selected_descriptor.keys_to_samples("species_center")
+selected_descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+
+properties = selected_descriptor.block().properties
+# get an array([0]) as a list of angular properties
+print(f"we have the following angular components: {np.unique(properties['l'])}")
+
+# The previous selection method uses the same selection for all blocks.
+# If you can to use different selection for different blocks, you should
+# use a `TensorMap` to create your selection
+
+selected_descriptor = calculator.compute(frames, selected_properties=selection)
+descriptor_for_comparison = calculator.compute(
+    frames, selected_properties=selected_descriptor
+)
+
+# The descriptor had 180 properties stored in the first block,
+# the selected_descriptor had 36. So descriptor_for_comparison
+# will also have 36 properties.
+print("shape of first block initially:", {descriptor.block(0).values.shape})
+print("shape of first block of reference:", {selected_descriptor.block(0).values.shape})
+print(
+    "shape of first block after selection:",
+    {descriptor_for_comparison.block(0).values.shape},
+)
+
+
+# The `TensorMap` format allows us to select features within
+# each block, and then construct a general matrix of features. We
+# can select the most significant features using FPS, which selects
+# features based on the  distance between them. The following algorithm
+# selects the 10 most important features in each block, then constructs
+# a new descriptor based on them, and calculates the final matrix of
+# features for it.
+
+# choose features:
+def fps_feature_selection(descriptor, n_to_select):
+    blocks = []
+    for _, block in descriptor:
+        # create a separate FPS selector for each block
+        fps = FPS(n_to_select=n_to_select)
+        mask = fps.fit(block.values).get_support()
+        selected_properties = block.properties[mask]
+        # put the selected features in a format rascaline can use
+        blocks.append(
+            TensorBlock(
+                # values, samples and component carry no information here
+                values=np.empty((1, len(selected_properties))),
+                samples=Labels.single(),
+                components=[],
+                properties=selected_properties,
+            )
+        )
+
+    return TensorMap(descriptor.keys, blocks)
+
+
+# We get `TensorMap` in the results of this function. The only thing that
+# is important in it is the properties, so we can create samples and
+# components empty altogether. In the case of values, we only need
+# form matching with len of properties.
+selection = fps_feature_selection(descriptor, n_to_select=10)
+# use the selection with rascaline
+selected_descriptor = calculator.compute(frames, selected_properties=selection)
+
+# Note that in this case it is no longer possible to have a single feature matrix,
+# because each block will have its own properties.
```

Comment:
```suggestion
# because each block will have its own set of properties.
```

also, we should actually be able to move keys around, but it would require changing the names/values of the properties from (n1, n2, l) to (q) and [0, 1, 2, 3, ..., 9] instead of the actual values. Not sure if it is worth doing in this how to?

Reply:
This is an interesting question. On the one hand, people may have such a problem and it would be good to show them the solution. On the other hand, the tutorial already looks a bit large, and it might be worth writing another tutorial looking at the extra propblems there.

Reply:
Sure, let's leave it for later!